### PR TITLE
feat: support dynamic browser name

### LINKED ISSUE #731: [FEATURE] Playwright does not support dynamic browser name
Playwright in typescript supports accessing the browser name with its string name like :
```ts
const playwright = require('playwright');

(async () => {
  for (const browserType of ['chromium', 'firefox', 'webkit']) {
    const browser = await playwright[browserType].launch();
    const context = await browser.newContext();
    const page = await context.newPage();
    await page.goto('http://whatsmyuseragent.org/');
    await page.screenshot({ path: `example-${browserType}.png` });
    await browser.close();
  }
})();
```
Whereas in python, it will errors as the playwright object is not subscriptable 
```console
TypeError: 'Playwright' object is not subscriptable 
```

Playwright should make the playwright subscriptable to be able to dynamically launch browser from string

TBD API:
```py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    for browser_type in ["chromium", "firefox", "webkit"]:
        browser = p[browser_type].launch()
        page = browser.new_page()
        page.goto("http://whatsmyuseragent.org/")
        page.screenshot(path=f"example-{browser_type.name}.png")
        browser.close()

```

### PR DESCRIPTION
**Pull request by kumaraditya303**
Fixes #731

### CODE REVIEW

File: playwright/_impl/_playwright.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
```
Comment (mxschmitt) [inline playwright/_impl/_playwright.py:15]:
```suggestion
```

File: playwright/_impl/_playwright.py [hunk 2]
```diff
@@ -40,6 +41,15 @@ def __init__(
...
         }
 
+    def __getitem__(self, value: str) -> BrowserType:
+        if value == "chromium":
+            return self.chromium
+        elif value == "firefox":
+            return self.firefox
+        elif value == "webkit":
+            return self.webkit
+        raise ValueError(f"Invalid browser {value!r}")
```
Comment (mxschmitt) [inline playwright/_impl/_playwright.py:51]: out of curiosity, what does the !r do again?

File: playwright/_impl/_playwright.py [hunk 3]
```diff
@@ -40,6 +41,15 @@ def __init__(
             for device in initializer["deviceDescriptors"]
         }
 
+    def __getitem__(self, value: str) -> BrowserType:
+        if value == "chromium":
+            return self.chromium
+        elif value == "firefox":
+            return self.firefox
+        elif value == "webkit":
```
Comment (mxschmitt) [inline playwright/_impl/_playwright.py:49]:
```suggestion
        if value == "firefox":
            return self.firefox
        if value == "webkit":
```

File: tests/assets/client.py [hunk 4]
```diff
@@ -12,22 +12,22 @@
...
-    for browser_type in [playwright.chromium, playwright.firefox, playwright.webkit]:
-        browser = browser_type.launch()
-        page = browser.new_page()
-        page.goto("https://example.com")
-        here = Path(__file__).parent.resolve()
-        page.screenshot(path=here / f"{browser_type.name}.png")
-        page.close()
-        browser.close()
+def main() -> None:
+    with sync_playwright() as p:
```
Comment (mxschmitt) [inline tests/assets/client.py:22]: having additionally a normal test-case in the browser tests for async and sync would be good to have.

**Review by mxschmitt — APPROVED** [review]
LGTM, just nits

File: playwright/_impl/_playwright.py (hunk 2 above)
Comment (kumaraditya303) [reply playwright/_impl/_playwright.py:51]: Prints the browser name in quotes which looks nice in traceback

File: tests/assets/client.py (hunk 4 above)
Comment (kumaraditya303) [reply tests/assets/client.py:22]: Okay

File: playwright/_impl/_playwright.py (hunk 3 above)
Comment (kumaraditya303) [reply playwright/_impl/_playwright.py:49]: else if should be fine I guess ?

Comment (mxschmitt) [reply playwright/_impl/_playwright.py:49]: yeah its more a personal preference.

File: tests/async/test_dynamic_browser_name.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from playwright.sync_api import sync_playwright
+
+
+def test_dynamic_browser_name(browser_name, launch_arguments):
+    with sync_playwright() as p:
+        browser = p[browser_name].launch(**launch_arguments)
+        page = browser.new_page()
+        page.goto("https://github.com")
```
Comment (mxschmitt) [inline tests/async/test_dynamic_browser_name.py:21]: We don't want to rely on external websites for the test. A simple assert page.evaluate("1 + 1") == 2 would be enough.

Comment (kumaraditya303) [reply tests/async/test_dynamic_browser_name.py]: Fixed

File: playwright/_impl/_playwright.py [hunk 6]
```diff
@@ -40,6 +40,15 @@ def __init__(
             for device in initializer["deviceDescriptors"]
         }
 
+    def __getitem__(self, value: str) -> BrowserType:
```
Comment (pavelfeldman) [inline playwright/_impl/_playwright.py:43]: I would inline this exact snippet when generating Playwright (both sync and async), as a single hard-coded blob. That way we don't grow complexity in the generator.

**Comment by pavelfeldman** [comment]
Where did the trailing coma come from in all the generated files?

**Comment by kumaraditya303** [comment]
Black formatted it

**Comment by pavelfeldman** [comment]
> Black formatted it

I don't see where you are changing the Black settings.

**Comment by kumaraditya303** [comment]
> > Black formatted it
> 
> I don't see where you are changing the Black settings.

Even I don't know, https://test-black.readthedocs.io/en/style-guide/style_guide/trailing_commas.html explains it

**Comment by kumaraditya303** [comment]
@pavelfeldman As you can see here in the diff black is adding the comma https://github.com/microsoft/playwright-python/pull/748/checks?check_run_id=2914309332 and with the diff the CI passes hence not wating more time Can you merge this ?

**Comment by pavelfeldman** [comment]
This looks like a Black bug: if you replace `f"Invalid browser {value!r}"` with `"Invalid browser " + value` it goes back to normal.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**